hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -324,7 +324,7 @@ public HazelcastProperties getProperties() {
         T service = serviceManager.getService(serviceName);
         if (service == null) {
             if (isRunning()) {
-                throw new HazelcastException("Service with name '" + serviceName + "' not found!");
+                throw new ServiceNotFoundException("Service with name '" + serviceName + "' not found!");
```

Comment:
I had discussed it with @mustafaiman and I will throw `ServiceNotFoundException` as cause of `HazelcastException`.